### ISSUE TITLE
Permission problem for ServiceWorker

### ISSUE THREAD

**Issue by isagalaev** [mozilla/send#1222]
Getting this error from the main page:

```
Failed to register/update a ServiceWorker for scope ‘https://send.firefox.com/’: Storage access is restricted in this context due to user settings or private browsing mode. main.js:38:10
```

Using Firefox 65 in normal non-private mode on Ubuntu with the strict content blocking policy (although I don't believe it's causing the problem). It may be caused by some leftover setting in my ancient user profile but I have no idea where to look. Any pointers?

**Comment by philipp-sumo** [comment]
hi, did you set "Delete cookies and site data when Firefox is closed" in your firefox settings?
similar cases got reported on bugzilla too: 
https://bugzilla.mozilla.org/show_bug.cgi?id=1534680
https://bugzilla.mozilla.org/show_bug.cgi?id=1534750
https://bugzilla.mozilla.org/show_bug.cgi?id=1534892

**Comment by isagalaev** [comment]
I do indeed. And adding `https://send.firefox.com` to exceptions does fix the problem, thanks. Feel free to close this issue or dupe it against any of the other ones you see fit.

**Comment by johngruen** [comment]
Thanks @isagalaev we have a fix that will resolve this without adding exceptions. We'll leave this bug open until that gets deployed. Thanks for filing!

**Comment by isagalaev** [comment]
I was actually wondering how it might be fixed in general. My gut tells me it might actually be a bug in Firefox if it doesn't want to give permissions to create a ServiceWorker even for the duration of a session. Or may be it's something in the feature detection code on the site?

**Comment by SoftVision-PaulOiegas** [comment]
@isagalaev that was my hunch too [here](https://github.com/mozilla/send/issues/1226#issuecomment-472410396) where I performed some deeper investigations.

**Comment by dannycoates** [comment]
Here's the bug to follow on this: https://bugzilla.mozilla.org/show_bug.cgi?id=1413615

TLDR; According to comment 14 `we can/should indeed "allow service workers when cookie policy is 'keep until I close firefox'".`

**Comment by RomanEmperor** [comment]
> hi, did you set "Delete cookies and site data when Firefox is closed" in your firefox settings?
> similar cases got reported on bugzilla too:
> https://bugzilla.mozilla.org/show_bug.cgi?id=1534680
> https://bugzilla.mozilla.org/show_bug.cgi?id=1534750
> https://bugzilla.mozilla.org/show_bug.cgi?id=1534892

UnChecking "Delete cookies and site data when Firefox is closed" in firefox settings?
Worked for me
